### ISSUE TITLE
Factoring out OAuth code/helping improve OAuth.jl

### ISSUE THREAD

**Issue by randyzwitch** [quinnj/Yelp.jl#1]
Hey Jacob - 

I finally got around to stripping the OAuth code from the Twitter package. After reading RFC5849 and the liboauth C library definition, I just did my interpretation of the standard via re-factoring my code.

https://github.com/randyzwitch/OAuth.jl

Any chance you could try my code out in your scenario? Before releasing to METADATA, I'd love to have two working examples to prove that the package works (since I'm flying blind on my interpretation!)

**Comment by randyzwitch** [comment]
I just published OAuth to METADATA...I think I'm going to try and substitute the library in for your code, so I can see where OAuth.jl needs updating and figure out how the Yelp API works :)

**Comment by quinnj** [comment]
Yeah, sorry for the delay here. I just started a new job, so my Julia time has unfortunately been limited lately.

**Comment by randyzwitch** [comment]
No problem at all, I wasn't waiting on you (and I saw your LinkedIn update, congrats on the new position!). I published OAuth to METADATA last night, because it was getting too hard for me to update my Twitter package while having a dependency on an unregistered package.

If I can get some work done, I'll just submit a PR with everything, since there aren't very many LOC in your package.
